Jetchat/app/src/main/java/com/example/compose/jetchat/NavActivity.kt

```diff
@@ -17,72 +17,98 @@
-        // Passing each menu ID as a set of Ids because each
-        // menu should be considered as top level destinations.
-        appBarConfiguration = AppBarConfiguration(
-            setOf(
-                R.id.nav_home,
-                R.id.nav_profile
-            ),
-            drawerLayout
-        )
-        navView.setupWithNavController(navController)
-    }
+                val openDrawerEvent = viewModel.drawerShouldBeOpened.observeAsState()
+                if (openDrawerEvent.value == true) {
+                    // Open drawer and reset state in VM.
+                    scaffoldState.drawerState.open {
+                        viewModel.resetOpenDrawer()
+                    }
+                }
 
-    override fun onCreateOptionsMenu(menu: Menu): Boolean {
-        // Inflate the menu; this adds items to the action bar if it is present.
-        menuInflater.inflate(R.menu.main, menu)
-        return true
-    }
+                // Intercepts back navigation when the drawer is open
+                backPressHandler(
+                    enabled = scaffoldState.drawerState.isOpen,
+                    onBackPressed = { scaffoldState.drawerState.close() },
+                    highPriority = true
+                )
 
-    override fun onSupportNavigateUp(): Boolean {
-        val navController = findNavController(R.id.nav_host_fragment)
-        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
+                JetchatScaffold(scaffoldState)
+            }
+        }
     }
 
-    /**
-     * Back closes drawer if open.
-     */
-    override fun onBackPressed() {
-        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
-            drawerLayout.closeDrawer(GravityCompat.START)
-        } else {
-            super.onBackPressed()
+    @Composable
+    private fun JetchatScaffold(
+        scaffoldState: ScaffoldState,
+    ) {
+        JetchatTheme {
+            Scaffold(
+                scaffoldState = scaffoldState,
+                drawerContent = {
+                    JetchatDrawer(
+                        onChatClicked = {
+                            findNavController(R.id.nav_host_fragment)
+                                .popBackStack(R.id.nav_home, true)
+                            scaffoldState.drawerState.close()
+                        },
+                        onProfileClicked = {
+                            val bundle = bundleOf("userId" to it)
+                            findNavController(R.id.nav_host_fragment).navigate(
+                                R.id.nav_profile,
+                                bundle
+                            )
+                            scaffoldState.drawerState.close()
+                        }
+                    )
+                }
+            ) {
+                val context = ContextAmbient.current
+
+                // Draw the
+                AndroidView({
+                    LayoutInflater.from(context)
```

Comment:
Can this layout use ViewBinding? eng is really against the pattern of calling `LayoutInflater` inside the `AndroidView` lambda